controller/api_impl/base_entity_api_model.go

```diff
@@ -0,0 +1,122 @@
+/*
```

Comment:
Mmmmm again lots of code duped from the edge. Link builders and other base entity items should be consolidated over time.

Reply:
Agree. I did a first pass at reducing duplication, but there's lots left to do